### PR TITLE
skawarePackages: fix llvm build

### PR DESCRIPTION
**Pull request by kampka**
###### Motivation for this change

skaware packages like execline do not build correctly with llvm.
Basically all skawarePackages face this issue, so it makes sense to fix it in the build wrapper.

```
nix build nixpkgs#pkgsLLVM.skawarePackages.execline
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Profpatsch** [comment]
The reformat was slightly unnecessary, but looks good to me.

**Comment by kampka** [comment]
> The reformat was slightly unnecessary, but looks good to me.

Agreed from a technical point, but ofBorg complained about some whitespace being wrong and I am too old to find joy in doing menial tasks manually that a machine can do for me.

**Comment by Profpatsch** [comment]
> , but ofBorg complained about some whitespace being wrong

now that’s just silly.

File: pkgs/build-support/skaware/build-skaware-package.nix [hunk 1]
```diff
@@ -73,19 +74,21 @@ in stdenv.mkDerivation {
...
     # Explicitly setting target ensures code can be compiled against a skalibs
     # binary built on a different version of darwin.
     # http://www.skarnet.org/cgi-bin/archive.cgi?1:mss:623:heiodchokfjdkonfhdph
     ++ (lib.optional stdenv.isDarwin
-         "--build=${stdenv.hostPlatform.system}");
+    "--build=${stdenv.hostPlatform.system}");
 
   inherit postConfigure;
 
+  makeFlags = lib.optional stdenv.cc.isClang [ "AR=${stdenv.cc.targetPrefix}ar" "RANLIB=${stdenv.cc.targetPrefix}ranlib" ];
```
Comment (alyssais) [inline pkgs/build-support/skaware/build-skaware-package.nix:90]: Why not apply this unconditionally?  The more conditions we have, the more difficult the code becomes to read and maintain.

**Pull request merged**